app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -644,30 +645,51 @@ export const simulatePersonalSign = async (params: PersonalSignParams | SignType
 	throw new Error(`Simulated signing not implemented for method ${ params.method }`)
 }
 
-const getSimulatedTokenBalances = async (ethereumClientService: EthereumClientService, transactionQueue: EthereumUnsignedTransaction[], balances: { token: bigint, owner: bigint }[], blockNumber: bigint): Promise<TokenBalancesAfter> => {
+type BalanceQuery = {
+	type: 'ERC20',
+	token: bigint,
+	owner: bigint,
+} | {
+	type: 'ERC1155',
+	token: bigint,
+	owner: bigint,
+	tokenId: bigint,
+}
+
+const getSimulatedTokenBalances = async (ethereumClientService: EthereumClientService, transactionQueue: EthereumUnsignedTransaction[], balances: BalanceQuery[], blockNumber: bigint): Promise<TokenBalancesAfter> => {
 	if (balances.length === 0) return []
-	const tokenInterface = new ethers.Interface(['function balanceOf(address account) view returns (uint256)'])
+	const erc20TokenInterface = new ethers.Interface(['function balanceOf(address account) view returns (uint256)'])
+	const erc1155TokenInterface = new ethers.Interface(['function balanceOf(address _owner, uint256 _id) external view returns(uint256)'])
 	const transactions = balances.map((balanceRequest, index) => {
-		const balanceOfCallData = stringToUint8Array(tokenInterface.encodeFunctionData('balanceOf', [addressString(balanceRequest.owner)]))
-		return {
+		const base = {
 			type: '1559' as const,
 			from: MOCK_ADDRESS + BigInt(index) + 1n,
 			to: balanceRequest.token,
 			value: 0n,
-			input: balanceOfCallData,
 			maxFeePerGas: 0n,
 			maxPriorityFeePerGas: 0n,
 			accessList: [],
 			gas: 42000n,
 			chainId: 0n,
 			nonce: 0n,
 		}
+		if (balanceRequest.type === 'ERC20') {
+			return {
+				...base,
+				input: stringToUint8Array(erc20TokenInterface.encodeFunctionData('balanceOf', [addressString(balanceRequest.owner)])),
+			}
+		}
+		return {
+			...base,
+			input: stringToUint8Array(erc1155TokenInterface.encodeFunctionData('balanceOf', [addressString(balanceRequest.owner), balanceRequest.tokenId])),
+		}
 	})
 	const transactionQueueSize = transactionQueue.length
 	const response = await ethereumClientService.multicall(transactionQueue.concat(transactions), blockNumber)
 	if (response.length !== transactions.length + transactionQueueSize) throw new Error('Multicall length mismatch')
 	return balances.map((balance, index) => ({
 		token: balance.token,
+		tokenId: 'tokenId' in balance ? balance.tokenId : undefined,
```

Comment:
You might be checking for type here?
```suggestion
		tokenId: balance.type === 'ERC1155' ? balance.tokenId : undefined,
```

Reply:
what do you mean by that I might be checking for type?